include/core/Discretization.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+struct Discretization
+{
+    virtual ~Discretization() = default;
+
+    enum GridType { hydrofabric_locations, spatial_structured, spatial_unstructured };
+    virtual GridType grid_type() = 0;
+
+    // Must cover an identified geographic domain
+    virtual GeoDivide divide() = 0;
```

Comment:
Or maybe omit this, if we'd have to get into weird constructive geometry expressions, like "hydro-locations across the CONUS domain, minus those within a region of interest modeled by a different formulation"